src/include/error.h

```diff
@@ -143,32 +159,27 @@
-                                                                                                \
-        /*                                                                                      \
-         * Normally these are two errx messages, but we need to stitch them both together into  \
-         * unittest_assert_msg. To do so insert the second message immediately after the first. \
-         */                                                                                     \
-        WT_IGNORE_RET(__wt_snprintf_len_set((session)->unittest_assert_msg,                     \
-          WT_SESSION_UNITTEST_BUF_LEN, &_offset, "Assertion '%s' failed: ", #exp));             \
-        WT_IGNORE_RET(__wt_snprintf((session)->unittest_assert_msg + _offset,                   \
-          WT_SESSION_UNITTEST_BUF_LEN - _offset, __VA_ARGS__));                                 \
-                                                                                                \
-        (session)->unittest_assert_hit = true;                                                  \
+#define TRIGGER_ABORT(session, exp, ...)                                                    \
+    do {                                                                                    \
+        if ((session) == NULL) {                                                            \
+            __wt_errx(                                                                      \
+              session, "A non-NULL session must be provided when unit testing assertions"); \
+            __wt_abort(session);                                                            \
+        }                                                                                   \
+        /*                                                                                  \
+         * For Atlas log parsing purposes.                                                  \
+         */                                                                                 \
+        BUILD_ASSERTION_STRING(                                                             \
+          session, (session)->unittest_assert_msg, WT_ERR_MSG_BUF_LEN, exp, __VA_ARGS__);   \
+        (session)->unittest_assert_hit = true;                                              \
     } while (0)
 #else
-#define TRIGGER_ABORT(session, exp, ...)                     \
-    do {                                                     \
-        __wt_errx(session, "Assertion '%s' failed: ", #exp); \
-        __wt_errx(session, __VA_ARGS__);                     \
-        __wt_abort(session);                                 \
+#define TRIGGER_ABORT(session, exp, ...)                                             \
+    do {                                                                             \
+        char _buf[WT_ERR_MSG_BUF_LEN];                                               \
```

Comment:
Do we need the unittest global buffer if we have one localised now?

Reply:
There are two different definitions of `trigger_abort`, one for when we're unit testing and the other when we're not. The buffer you're referring to (`unittest_assert_msg`) is used specifically when unit testing to identify when the assertion has "fired".